rpmbuild/copr_rpmbuild/automation/fedora_review.py

```diff
@@ -45,8 +46,10 @@ def run(self):
             "--mock-config", self.mock_config_file,
         ]
 
+        os.makedirs(self.cachedir, exist_ok=True)
```

Comment:
I fear directory leftovers. Could you please use this pattern?
```
os.makedirs(self.cachedir, exist_ok=True)
try:
    ... other logic ...
finally:
    shutil.rmtree(self.cachedir)
```

I really like context managers for this:
```python
@contextmanater
def with_cache_dir(...):
    create_the_directory
    try:
        yield
    finally:
        drop_the_dir
```

Reply:
I've had a go at porting this over to a context manager. I'm not wild about having to change the signature of `_filter_results_directory`, but it felt cleaner than adding it to the `FedoraReview` class. Happy to do it that way if you'd prefer.